jwst/outlier_detection/imaging.py

```diff
@@ -82,37 +89,38 @@ def detect_outliers(
     else:
         # for non-dithered data, the resampled image is just the original image
         drizzled_models = input_models
-        for i in range(len(input_models)):
-            drizzled_models[i].wht = build_driz_weight(
-                input_models[i],
-                weight_type=weight_type,
-                good_bits=good_bits)
-        # copy for when saving median and input is a filename?
-        median_wcs = copy.deepcopy(input_models[0].meta.wcs)
+        with input_models:
+            for i, model in enumerate(input_models):
+                model.wht = build_driz_weight(
+                    model,
+                    weight_type=weight_type,
+                    good_bits=good_bits)
+                # copy for when saving median and input is a filename?
+                if i == 0:
+                    median_wcs = copy.deepcopy(model.meta.wcs)
 
     # Perform median combination on set of drizzled mosaics
-    median_data = create_median(drizzled_models, maskpt)
+    on_disk = not in_memory
+    median_data = create_median_library(drizzled_models, maskpt, on_disk=on_disk)
 
     if save_intermediate_results:
         # make a median model
-        with datamodels.open(drizzled_models[0]) as dm0:
-            median_model = datamodels.ImageModel(median_data)
-            median_model.update(dm0)
-            median_model.meta.wcs = median_wcs
+        with drizzled_models:
+            example_model = drizzled_models.borrow(0)
+            drizzled_models.shelve(example_model, modify=False)
+            with datamodels.open(example_model) as dm0:
+                median_model = datamodels.ImageModel(median_data)
+                median_model.update(dm0)
+                median_model.meta.wcs = median_wcs
+            del example_model
 
         save_median(median_model, make_output_path, asn_id)
         del median_model
-    else:
-        # since we're not saving intermediate results if the drizzled models
-        # were written to disk, remove them
-        if not in_memory:
-            for fn in drizzled_models._models:
-                remove_file(fn)
```

Comment:
I think this is still needed. I haven't run the local test but does outlier detection now leave behind these files after a run? Looking at the resample changes it looks like it takes a list of filenames, makes a fake association and creates a `ModelLibrary` with these files. Since the library won't touch the input files, is there code somewhere that deletes them (if `save_intermediate_results=False`)?

Reply:
you are right it was still needed in order to remove the outlier_s2d files